### PR TITLE
Factor out C++ specific options in `Options > Packages > C / C++`

### PR DESCRIPTION
**Pull request by romainfrancois**
### Intent

related to #10337

### Approach

image

### Automated Tests

> Indicate whether this change includes unit tests or integration tests, or link a work item or pull request to add those tests to another repo. If the change cannot or will not be covered by a test, indicate why.

### QA Notes

- Global options > Packages > C / C++ 
- select cpp11 as the template to use
- apply

- File > New File > New C++ file : this should create a file with: 

```cpp
#include <cpp11.hpp>

// This is a simple example of exporting a C++ function to R. You can
// source this function into an R session using cpp11::cpp_source()
// Learn more about cpp11 at: 
//
//   https://cpp11.r-lib.org/index.html

[[cpp11::register]]
double sum(cpp11::doubles x) {
    double total = 0.0;
    for (double value: x) {
        total += x;
    }
    return total;
}
```

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

**Comment by romainfrancois** [comment]
This does not include something to override the default setting based on `LinkingTo: `

File: src/cpp/session/resources/templates/cpp11.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include <cpp11.hpp>
+
+// This is a simple example of exporting a C++ function to R. You can
+// source this function into an R session using cpp11::cpp_source()
```
Comment (kevinushey) [inline src/cpp/session/resources/templates/cpp11.cpp:4]:
```suggestion
// source this function into an R session using `cpp11::cpp_source()`.

```

**Comment by romainfrancois** [comment]
That's better, and I'm sure we'll come up with a few additional options later. 

image

File: src/gwt/src/org/rstudio/studio/client/workbench/prefs/PrefsConstants_en.properties [hunk 2]
```diff
@@ -213,7 +213,8 @@ developmentNavigateLabel=Automatically navigate editor to build errors
 developmentHideLabel=Hide object files in package src directory
 developmentCleanupLabel=Cleanup output after successful R CMD check
 developmentViewLabel=View Rcheck directory after failed R CMD check
-developmentRcppLabel=Use Rcpp template when creating C++ files
+developmentCppTemplate=Use template for new C++ files
```
Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/workbench/prefs/PrefsConstants_en.properties:216]:
```suggestion
developmentCppTemplate=Use template for new C++ files:
```

For menu box labels, we typically end the label with a colon.

File: src/gwt/src/org/rstudio/studio/client/workbench/prefs/PrefsConstants_fr.properties [hunk 3]
```diff
@@ -213,7 +213,8 @@ developmentNavigateLabel=Navigation automatique de l''éditeur vers les erreurs
 developmentHideLabel=Cacher les fichiers objets dans le répertoire src du paquet
 developmentCleanupLabel=Nettoyage de la sortie après une vérification R CMD réussie
 developmentViewLabel=Affichage du répertoire Rcheck après l''échec de la vérification R CMD
-developmentRcppLabel=Utiliser le modèle Rcpp lors de la création de fichiers C++.
+developmentCppTemplate=Modèle à utiliser lors de la création de fichiers C++
```
Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/workbench/prefs/PrefsConstants_fr.properties:216]:
```suggestion
developmentCppTemplate=Modèle à utiliser lors de la création de fichiers C++:
```

**Review by kevinushey — APPROVED** [review]

**Comment by jgutman** [comment]
This is only available in the Global Options right, not in Project Options? Would it make sense as a follow up to have this available in Project Options as well, in case a user wanted a project-specific override in something that is not a package, or even a project-specific override that might contradict what is suggested by `LinkingTo`?

**Comment by romainfrancois** [comment]
Probably not, once you've selected which backend `Rcpp` or `cpp11` in the `DESCRIPTION` you rarely/never mix.

**Pull request merged**